dbt/include/athena/macros/adapters/metadata.sql

```diff
@@ -65,9 +65,23 @@
 
                 )
 
-                {%- for schema in schemas -%}
-                select * from catalog where lower("table_schema") = lower('{{ schema }}')
-                {%- if not loop.last %} union all {% endif -%}
+                {%- for db, tables in schemas.items() -%}
+                  {%- for table_batch in tables|batch(100) %}
+                    select * from catalog where "table_schema" = lower('{{ db }}') 
```

Comment:
Just a suggestion but if we wanted to keep naming conventions the same as the current code we could replace `db` as `schema`
```suggestion
                {%- for schema, tables in schemas.items() -%}
                  {%- for table_batch in tables|batch(100) %}
                    select * from catalog where "table_schema" = lower('{{ schema }}') 
```

Reply:
👍  Renamed `tables` to `relations` as well